torch/_dynamo/trace_rules.py

```diff
@@ -63,11 +63,36 @@
     "torch.distributed.get_rank": TorchInGraphFunctionVariable,
     "torch.distributed.get_world_size": TorchInGraphFunctionVariable,
     "torch.distributed._tensor.DTensor#from_local": TorchInGraphFunctionVariable,
+    "torch.distributed.distributed_c10d._get_group_tag": TorchInGraphFunctionVariable,
+    "torch.distributed.distributed_c10d.get_process_group_ranks": TorchInGraphFunctionVariable,
     "torch._utils.is_compiling": TorchInGraphFunctionVariable,
     "torch.overrides.get_default_nowrap_functions": TorchInGraphFunctionVariable,
     "torch.fx._symbolic_trace.is_fx_tracing": TorchInGraphFunctionVariable,
     "torch._dynamo.external_utils.is_compiling": TorchInGraphFunctionVariable,
     "torch.autograd.graph.disable_saved_tensors_hooks": TorchInGraphFunctionVariable,
+    "torch.autograd._profiler_enabled": SkipFilesVariable,
+    # We graph break on RNG state setters or getters like
+    # `torch.get_rng_state` or `torch.set_rng_state`. These functions
+    # are not aten operations and therefore they are completely ignored
+    # by the AOT dispatcher. As a result, the AOT graph does not have
+    # these setter or getter functions, producing an incorrect graph
+    # when it comes to rng states.
+    "torch.default_generator#get_state": SkipFilesVariable,
+    "torch._C.Generator#get_state": SkipFilesVariable,
+    "torch.get_rng_state": SkipFilesVariable,
+    "torch.cuda.get_rng_state": SkipFilesVariable,
+    "torch.default_generator#set_state": SkipFilesVariable,
+    "torch._C.Generator#set_state": SkipFilesVariable,
+    "torch.set_rng_state": SkipFilesVariable,
+    "torch.cuda.set_rng_state": SkipFilesVariable,
+    # https://github.com/pytorch/pytorch/issues/107187
+    "torch.manual_seed": SkipFilesVariable,
+    # https://github.com/pytorch/pytorch/issues/93501
+    "torch.nn.utils.rnn.pack_padded_sequence": SkipFilesVariable,
+    # https://github.com/pytorch/pytorch/issues/99569
+    "torch.nn.Parameter": SkipFilesVariable,
+    "torch._nested_tensor_from_mask": SkipFilesVariable,
+    "torch._nested_from_padded": SkipFilesVariable,
```

Comment:
There functions were handled by many ```elif ...: unimplemented(...)``` in torch.py originally, I think adding an item  ```xxx: SkipFilesVariable``` to ```manual_torch_name_rule_map``` is the best way if we want to graph break on a specific function going forward.
Inside of ```TorchInGraphFunctionVariable.call_function```, we only need to handle these functions which are real in graph functions.

Reply:
We should have mechanism to ensure there is no duplications across all of these dicts:
* E.g, if we indicate ```torch.manual_seed``` as skipping, we should remove it from the in_graph_function list.
* The different names pointing to the same object, e.g, ```torch.manual_seed``` is ```torch.random.manual_seed```.

I'll have a separate PR of adding other unit tests to ensure these lists are updated correctly.